### PR TITLE
Added extra option to enable the Snow Halation theme

### PR DESCRIPTION
**Pull request by ramzialhaddad**
Adds an extra option to the SecretRingToneEnabler plugin to only play the Snow Halation discord ring theme.

Let me know if there are any errors or discrepancies!

### CODE REVIEW

**Comment by sadan4** [comment]
i cant hear a difference, is your patch correct?

File: src/plugins/secretRingTone/index.ts [hunk 1]
```diff
@@ -4,14 +4,14 @@
  * SPDX-License-Identifier: GPL-3.0-or-later
  */
 
+import { Settings } from "@api/Settings";
 import { Devs } from "@utils/constants";
-import definePlugin from "@utils/types";
+import definePlugin, { OptionType } from "@utils/types";
 
-// NOTE - Ultimately should probably be turned into a ringtone picker plugin
```
Comment (sadan4) [inline src/plugins/secretRingTone/index.ts:10]: why did you remove this comment?

Comment (ramzialhaddad) [reply src/plugins/secretRingTone/index.ts:10]: Initially I checked if there was an existing plugin that did this PR functionality and I stumbled on #1827 which got moved to #1765. And as such I felt like the comment was then redundant. Have I made a mistake?

**Comment by ramzialhaddad** [comment]
> i cant hear a difference, is your patch correct?

It should be, I tested it both on a client and via the extension build. That is very odd.

**Comment by ramzialhaddad** [comment]
> i cant hear a difference, is your patch correct?

Just confirming too you did enable the setting "Only Snow" right? Inside the plugin settings window?

**Comment by sadan4** [comment]
not your fault, they were so similar that I didn't notice a difference until I googled them

sorry about that

**Comment by ramzialhaddad** [comment]
No worries

File: src/plugins/secretRingTone/index.ts [hunk 2]
```diff
@@ -20,5 +20,21 @@ export default definePlugin({
...
+        {
+            find: '"call_ringing_beat"',
+            predicate: ()=> Settings.plugins.SecretRingToneEnabler.onlySnow,
+            replacement: {
+                match: /"call_ringing_beat",/,
+                replace: "",
+            }
+        },
     ],
+    options: {
```
Comment (Nuckyz) [inline src/plugins/secretRingTone/index.ts:32]: Use definePluginSettings please

File: src/plugins/secretRingTone/index.ts [hunk 3]
```diff
@@ -20,5 +20,21 @@ export default definePlugin({
                 replace: "false",
             }
         },
+        {
+            find: '"call_ringing_beat"',
```
Comment (Nuckyz) [inline src/plugins/secretRingTone/index.ts:24]: Move this to the same find as the patch above and put the predicate inside the replacement

**Review by Nuckyz — CHANGES_REQUESTED** [review]

File: src/plugins/secretRingTone/index.ts (hunk 3 above)
Comment (ramzialhaddad) [reply src/plugins/secretRingTone/index.ts]: Completed

File: src/plugins/secretRingTone/index.ts (hunk 2 above)
Comment (ramzialhaddad) [reply src/plugins/secretRingTone/index.ts]: Done

File: src/plugins/secretRingTone/index.ts [hunk 4]
```diff
@@ -4,21 +4,38 @@
...
 
+import { definePluginSettings } from "@api/Settings";
 import { Devs } from "@utils/constants";
-import definePlugin from "@utils/types";
+import definePlugin, { OptionType } from "@utils/types";
+
+const settings = definePluginSettings({
+    onlySnow: {
+        type: OptionType.BOOLEAN,
+        description: "Only play the Snow Halation Theme?",
```
Comment (Nuckyz) [inline src/plugins/secretRingTone/index.ts:14]:
```suggestion
        description: "Only play the Snow Halation Theme",
```

**Review by Nuckyz — CHANGES_REQUESTED** [review]

**Comment by ramzialhaddad** [comment]
Assuming there isn't anything else to be changed I believe this PR is ready? (First time PR-ing here :D)

**Comment by Nuckyz** [comment]
> Assuming there isn't anything else to be changed I believe this PR is ready? (First time PR-ing here :D)

yepp!

**Comment by Nuckyz** [comment]
Thanks for the contribution!

**Pull request merged**